server/src/test/java/org/elasticsearch/index/mapper/NestedObjectMapperTests.java

```diff
@@ -625,9 +625,12 @@ public void testParentObjectMapperAreNested() throws Exception {
     }
 
     public void testLimitNestedDocsDefaultSettings() throws Exception {
-        Settings settings = Settings.builder().build();
+        Settings settings = Settings.builder()
+            // make sure we're not hitting the total field limit
+            .put(MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey(), Long.MAX_VALUE)
```

Comment:
out of curiosity: why was this change required?

Reply:
Thanks for this question: Looking into this again made me realize that my code had a bug. There can be multiple dynamic updates for the same field (which is only merged later), and this test runs exactly into that scenario (where there are nested docs that all have the same field, and there are more than 1000 of them). This wrongly triggers the field count check (as all fields are the same). Except for interleaving the process of merging different field definitions with the parsing code (i.e. with `DocumentParser.createDynamicUpdate`), we need to relax the check a bit here and can only count the number of different field names. This is fixed in 12de303816d

Reply:
Urgs, had to follow this up with another commit: cd79731